modulemd/include/private/modulemd-module-private.h

```diff
@@ -174,4 +175,21 @@ modulemd_module_upgrade_streams (ModulemdModule *self,
                                  ModulemdModuleStreamVersionEnum mdversion,
                                  GError **error);
 
+/**
+ * modulemd_module_associate_upgrade_helper:
+ * @self: This #ModulemdModule object.
+ * @upgrade_helper: (in)(transfer none): a #ModulemdUpgradeHelper to associate
+ * with this module.
+ *
+ * Associates a #ModulemdUpgradeHelper, usually from a #ModulemdModuleIndex
+ * with this module to help with upgrades between #ModuleStreamV2 and
+ * #ModuleStreamV3. This function will take a reference on the one passed in,
+ * so modifications to the @upgrade_helper will affect this #ModulemdModule.
```

Comment:
```suggestion
 * so modifications to the @upgrade_helper will affect subsequent changes to this
 * #ModulemdModule.
```

Just to be clear that changing @upgrade_helper won't affect any StreamV2s that have already been upgraded to StreamV3.